Assets/Shopify/UIToolkit/Editor/ProductPicker.cs

```diff
@@ -0,0 +1,155 @@
+﻿namespace Shopify.UIToolkit.Editor {
+    using System.Collections;
+    using System.Collections.Generic;
+    using UnityEngine;
+    using UnityEditor;
+    using Shopify.Unity;
+    using Shopify.Unity.SDK;
+    using System;
+    using System.Linq;
+
+    public interface IProductPickerView {
+        void DrawProductLoadingState(SerializedObject serializedObject);
+        void DrawProductPicker(SerializedObject serializedObject);
+        void DrawProductLoadingError(SerializedObject serializedObject);
+        void DrawShopHasNoProductsState(SerializedObject serializedObject);
+    }
+
+    public class ProductPicker : IProductPickerView {
+        private class ProductNameToGIDMapEntry {
+            public string ProductName;
+            public string ProductGID;
+        }
+
+        public IProductPickerView View;
+        private static List<ProductNameToGIDMapEntry> _cachedProductNameToGIDMap;
+        private List<ProductNameToGIDMapEntry> _productNameToGIDMap;
+        private ShopifyClient _client;
+        private ShopifyError _error;
+
+        public bool ProductsFinishedLoading() {
+            return _productNameToGIDMap != null || _error != null;
+        }
+
+        public ProductPicker(ShopifyClient client) {
```

Comment:
Forgot to mention - This feels a bit weird, assigning this to the View property here. Why not extract it into a ProductPickerView object, passing in a IProductPickerView instance that has a default value of a new ProductPickerView instance:

```csharp
public ProductPicker(ShopifyClient client, IProductPickerView view = new ProductPickerView()) {
    // where the magic happens
}
```